src/api/v2/types.rs

```diff
@@ -335,3 +356,36 @@ pub fn handle_result(result: Result<impl Reply, impl Reply>) -> impl Reply {
 		Err(err) => err.into_response(),
 	}
 }
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "topic", rename_all = "kebab-case")]
+pub enum WsResponse {
+	Error(Error),
+	Version(Response<Version>),
+	Status(Response<Status>),
+	DataTransactionSubmitted(Response<SubmitResponse>),
+}
```

Comment:
I think error is not a response. Error is an error. `Result` implements `Serialize` and `Deserialize` I think, we can just return `Result<WsResponse, Error>` instead.

Or as an alternative we can create a wrapper around the result which would return [untagged](https://serde.rs/enum-representations.html#untagged) result, so that we won't need to change API.

Reply:
I will check, but keep in mind that this error is just another web socket message sent to client. Maybe we can represent both server and client errors with the same abstraction and use it in control flow, but Ill have to think about it.

Reply:
Sure! What I mean is that it would be cool to just have error prone type separate from error and use `Result` as much as possible